azure/util.py

```diff
@@ -0,0 +1,309 @@
+import argparse
+import yaml
+import json
+import os
+from os import walk
+import os.path
+import subprocess
+import uuid
+import urllib.request
+import tarfile
+import zipfile
+from shutil import copyfile
+
+PETCTL_DIR = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+
+
+# This method runs all commands in a separate
+# process and returns the output
+def run_commands(cmds):
+    output = []
+    set_kubeconfig_environment_var()
+    
+    for cmd in cmds:
+        print("Running {}".format(cmd))
+        process = subprocess.Popen(
```

Comment:
In general, without `process.communicate`, the process might not start and output may not be available in `stdout` and `stderr`. I suggest using `subprocess.Run` that returns a `CompletedProcess` object containing the return code, stderr and stdout. Also, you might want to set `text=True` to make sure the return values are `str` objects and not `byte`s.

Reply:
Changed it to subprocess.run instead of Popen. Python 3.6 doesn't support text=True, the output is in the form of string currently.